MetaMask/metamask-extension
# Allow multiple Ledger & Trezor hardware accounts

Fix:
- https://github.com/MetaMask/metamask-extension/issues/9045
- https://github.com/MetaMask/metamask-extension/issues/6649
- https://github.com/MetaMask/metamask-extension/issues/10333

Explanation:  

This PR will allow users to add multiple hardware wallet accounts and keep them for any given amount of time.  While that in itself is awesome, this should also prevent the need for users to re-add tokens each time they use a different hardware wallet account.

Manual testing steps:  
  - Click "Connect Hardware Wallet", unlock your Ledger, choose "Ledger", click multiple address checkboxes, and Add them
  - You should see the accounts you had selected
  - Send ETH between them, use the test dapp to sign, etc.
  - Remove individual accounts, ensure correct account is removed
  - (Do the same with Trezor)

To do:

 - [x] Disable checkboxes in the account listing for addresses that have already been added
 - [x] Remove the need for `setTimeout`
 - [ ] What do we want to do re: naming if the user has two Ledgers and add two accounts with the same index?

This PR will require an update from both of our keyring repos:
- Ledger: https://github.com/MetaMask/eth-ledger-bridge-keyring/pull/48
- Trezor: https://github.com/MetaMask/eth-trezor-keyring/pull/54
